--- digikam-4.9.0/extra/kipi-plugins/CMakeLists.txt.orig	2015-04-08 22:28:07.106116273 +0100
+++ digikam-4.9.0/extra/kipi-plugins/CMakeLists.txt	2015-04-08 22:28:27.438117621 +0100
@@ -38,7 +38,7 @@
 
 # NOTE: This string is used to set libkipiplugins SO version ID
 set(KIPIPLUGINS_LIB_SO_VERSION_STRING
-    "${KIPIPLUGINS_MAJOR_VERSION}.${KIPIPLUGINS_MINOR_VERSION}.${KIPIPLUGINS_PATCH_VERSION}"
+    "${KIPIPLUGINS_MAJOR_VERSION}"
    )
 
 # =======================================================
--- digikam-4.9.0/core/app/CMakeLists.txt.orig	2015-04-08 22:26:30.715109881 +0100
+++ digikam-4.9.0/core/app/CMakeLists.txt	2015-04-08 22:27:53.634115380 +0100
@@ -125,7 +125,7 @@
    target_link_libraries(digikamcore ${KGEOMAP_LIBRARIES})
 endif()
 
-set_target_properties(digikamcore PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_VERSION_SHORT})
+set_target_properties(digikamcore PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_MAJOR_VERSION})
 
 if(WIN32)
     set_target_properties(digikamcore PROPERTIES COMPILE_FLAGS -DJPEG_STATIC)
@@ -184,7 +184,7 @@
     target_link_libraries(digikamdatabase ${SQLITE2_LIBRARIES})
 endif()
 
-set_target_properties(digikamdatabase PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_VERSION_SHORT})
+set_target_properties(digikamdatabase PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_MAJOR_VERSION})
 
 install(TARGETS digikamdatabase ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
