diff -Naur digikam-4.2.0/core/CMakeLists.txt digikam-4.2.0.tpg/core/CMakeLists.txt
--- digikam-4.2.0/core/CMakeLists.txt	2014-08-03 23:25:59.000000000 +0000
+++ digikam-4.2.0.tpg/core/CMakeLists.txt	2014-08-18 10:46:47.307877545 +0000
@@ -103,7 +103,7 @@
 
 INCLUDE(MacroUtils)
 INCLUDE(MacroOpenCV)
-DETECT_OPENCV(2.4.6 core highgui imgproc)
+DETECT_OPENCV(2.4.9 core highgui objdetect contrib legacy imgproc)
 
 # check the libkdcraw version
 
diff -Naur digikam-4.2.0/extra/kipi-plugins/CMakeLists.txt digikam-4.2.0.tpg/extra/kipi-plugins/CMakeLists.txt
--- digikam-4.2.0/extra/kipi-plugins/CMakeLists.txt	2014-08-03 23:26:07.000000000 +0000
+++ digikam-4.2.0.tpg/extra/kipi-plugins/CMakeLists.txt	2014-08-18 10:51:37.235050312 +0000
@@ -119,7 +119,7 @@
 
 INCLUDE(MacroUtils)
 INCLUDE(MacroOpenCV)
-DETECT_OPENCV(2.4.6 core highgui objdetect contrib)
+DETECT_OPENCV(2.4.9 core highgui objdetect contrib legacy imgproc)
 
 # OpenMP detection requires CMake >= 2.6.3
 
diff -Naur digikam-4.2.0/extra/libkface/CMakeLists.txt digikam-4.2.0.tpg/extra/libkface/CMakeLists.txt
--- digikam-4.2.0/extra/libkface/CMakeLists.txt	2014-08-03 23:26:00.000000000 +0000
+++ digikam-4.2.0.tpg/extra/libkface/CMakeLists.txt	2014-08-18 10:48:52.798380220 +0000
@@ -47,7 +47,7 @@
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 
 INCLUDE(MacroOpenCV)
-DETECT_OPENCV(2.4.9 core highgui objdetect contrib)
+DETECT_OPENCV(2.4.9 core highgui objdetect contrib legacy imgproc)
 
 INCLUDE_DIRECTORIES(${OpenCV_INCLUDE_DIRS})
 
