diff -urN digikam-4.4.0/core/app/CMakeLists.txt digikam-4.4.0-patched/core/app/CMakeLists.txt
--- digikam-4.4.0/core/app/CMakeLists.txt	2014-10-08 23:32:45.000000000 +1100
+++ digikam-4.4.0-patched/core/app/CMakeLists.txt	2014-10-13 20:23:13.986686928 +1100
@@ -132,7 +132,7 @@
    TARGET_LINK_LIBRARIES(digikamcore ${liblqr_LIBS})
 ENDIF(LQR-1_FOUND)
 
-SET_TARGET_PROPERTIES(digikamcore PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_VERSION_SHORT})
+SET_TARGET_PROPERTIES(digikamcore PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_MAJOR_VERSION})
 
 IF(WIN32)
     SET_TARGET_PROPERTIES(digikamcore PROPERTIES COMPILE_FLAGS -DJPEG_STATIC)
@@ -197,7 +197,7 @@
     TARGET_LINK_LIBRARIES(digikamdatabase ${SQLITE2_LIBRARIES})
 ENDIF(SQLITE2_FOUND)
 
-SET_TARGET_PROPERTIES(digikamdatabase PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_VERSION_SHORT})
+SET_TARGET_PROPERTIES(digikamdatabase PROPERTIES VERSION ${DIGIKAM_VERSION_SHORT} SOVERSION ${DIGIKAM_MAJOR_VERSION})
 
 INSTALL(TARGETS digikamdatabase ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
diff -urN digikam-4.4.0/extra/kipi-plugins/CMakeLists.txt digikam-4.4.0-patched/extra/kipi-plugins/CMakeLists.txt
--- digikam-4.4.0/extra/kipi-plugins/CMakeLists.txt	2014-10-08 23:33:05.000000000 +1100
+++ digikam-4.4.0-patched/extra/kipi-plugins/CMakeLists.txt	2014-10-13 20:23:13.986686928 +1100
@@ -32,7 +32,7 @@
    
 # NOTE: This string is used to set libkipiplugins SO version ID
 SET(KIPIPLUGINS_LIB_SO_VERSION_STRING
-    "${KIPIPLUGINS_MAJOR_VERSION}.${KIPIPLUGINS_MINOR_VERSION}.${KIPIPLUGINS_PATCH_VERSION}"
+    "${KIPIPLUGINS_MAJOR_VERSION}"
    )
 
 # =======================================================
