diff -up digikam-5.9.0/core/app/CMakeLists.txt.omv~ digikam-5.9.0/core/app/CMakeLists.txt
--- digikam-5.9.0/core/app/CMakeLists.txt.omv~	2018-11-26 11:56:43.816361087 +0100
+++ digikam-5.9.0/core/app/CMakeLists.txt	2018-11-26 11:59:07.091852929 +0100
@@ -230,7 +230,7 @@ if(HAVE_OPENGL)
 endif()
 
 if(KF5Sane_FOUND)
-    target_link_libraries(digikamcore PUBLIC KF5::Sane)
+    target_link_libraries(digikamcore PUBLIC KF5::Sane -fuse-ld=lld)
 endif()
 
 if(KF5AkonadiContact_FOUND)
