diff -urN digikam-3.5.0/extra/kipi-plugins/panorama/importwizard/lastpage.cpp digikam-3.5.0-patched/extra/kipi-plugins/panorama/importwizard/lastpage.cpp
--- digikam-3.5.0/extra/kipi-plugins/panorama/importwizard/lastpage.cpp	2013-10-01 17:23:09.000000000 +1100
+++ digikam-3.5.0-patched/extra/kipi-plugins/panorama/importwizard/lastpage.cpp	2013-10-11 23:07:51.000000000 +1100
@@ -91,11 +91,14 @@
 
     QLabel *fileTemplateLabel = new QLabel(i18n("File name template:"), d->saveSettingsGroupBox);
     formatVBox->addWidget(fileTemplateLabel);
-    QString first = d->mngr->itemsList().front().fileName();
-    QString last = d->mngr->itemsList().back().fileName();
-//     QString file = "panorama";
-    QString file = QString("%1-%2").arg(first.left(first.lastIndexOf('.'))).arg(last.left(last.lastIndexOf('.')));
-    d->fileTemplateKLineEdit  = new KLineEdit(file, d->saveSettingsGroupBox);
+    QString file = "panorama";
+    if (!d->mngr->itemsList().isEmpty())
+    {
+        QString first = d->mngr->itemsList().front().fileName();
+        QString last = d->mngr->itemsList().back().fileName();
+        file = QString("%1-%2").arg(first.left(first.lastIndexOf('.'))).arg(last.left(last.lastIndexOf('.')));
+    }
+    d->fileTemplateKLineEdit = new KLineEdit(file, d->saveSettingsGroupBox);
     d->fileTemplateKLineEdit->setToolTip(i18n("Name of the panorama file (without its extension)."));
     d->fileTemplateKLineEdit->setWhatsThis(i18n("<b>File name template</b>: Set here the base name of the files that "
                                                 "will be saved. For example, if your template is <i>panorama</i> and if "
