diff -up digikam-8.6.0/core/libs/widgets/text/localizeselector.cpp.omv~ digikam-8.6.0/core/libs/widgets/text/localizeselector.cpp
--- digikam-8.6.0/core/libs/widgets/text/localizeselector.cpp.omv~	2025-03-22 00:58:02.159359044 +0100
+++ digikam-8.6.0/core/libs/widgets/text/localizeselector.cpp	2025-03-22 01:31:50.789900341 +0100
@@ -326,7 +326,7 @@ bool s_inlineTranslateString(const QStri
     }
     else
     {
-        error = QChar(trengine->error());
+        error = QString::number(trengine->error());
     }
 
     return false;
