diff -urw digikam-5.0.0-beta2/core/libs/rawengine/libraw/internal/libraw_x3f.cpp /root/rpmbuild/BUILD/digikam-5.0.0-beta2/core/libs/rawengine/libraw/internal/libraw_x3f.cpp
--- digikam-5.0.0-beta2/core/libs/rawengine/libraw/internal/libraw_x3f.cpp	2015-11-30 22:08:43.000000000 +0000
+++ /root/rpmbuild/BUILD/digikam-5.0.0-beta2/core/libs/rawengine/libraw/internal/libraw_x3f.cpp	2016-01-20 12:58:09.928225347 +0000
@@ -1246,7 +1246,7 @@
   x3f_image_data_t *ID = &DEH->data_subsection.image_data;
   x3f_huffman_t *HUF = ID->huffman;
 
-  int16_t c[3] = {offset,offset,offset};
+  int16_t c[3] = {static_cast<int16_t>(offset),static_cast<int16_t>(offset),static_cast<int16_t>(offset)};
   int col;
   bit_state_t BS;
   
